### PR TITLE
Add get and update user plugin preferences rpcs

### PR DESCRIPTION
**Pull request by gilwong00**
This adds two new endpoints `GetUserPluginPreferences` and `UpdateUserPluginPreferences` to the `UserService`

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 1]
```diff
@@ -53,6 +54,19 @@ message OrganizationUser {
   OrganizationRoleSource organization_role_source = 4;
 }
 
+message UserSelectedPlugin {
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/user.proto:57]: Is there an existing type for this? What is a user-selected plugin? Needs docs explaining what this is at the minimum

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 2]
```diff
@@ -53,6 +54,19 @@ message OrganizationUser {
...
+  // owner of the plugin
+  string owner = 1;
+  // name of the plugin
+  string name = 2;
+}
+
+message UserPluginPreference {
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/user.proto:64]: What is this? needs docs at the minimum

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 3]
```diff
@@ -53,6 +54,19 @@ message OrganizationUser {
...
+  string user_id = 1;
+  PluginLanguage selected_language = 2;
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/user.proto:66]: Asymmetry between `UserSelectedPlugin` having `Selected` in it vs. this not, likely could be explained by understanding what API is and is for via docs

**Review by bufdev — CHANGES_REQUESTED** [review]

File: proto/buf/alpha/registry/v1alpha1/user.proto (hunk 1 above)
Comment (gilwong00) [reply proto/buf/alpha/registry/v1alpha1/user.proto]: we can actually dont need this type anymore. Removed.

**Comment by gilwong00** [comment]
@bufdev cleaned up this PR and added more docs around the new types

File: proto/buf/alpha/registry/v1alpha1/user.proto (hunk 2 above)
Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/user.proto:60]: That is this used for? The documentation just says what the fields are, I'm not sure what this structure is for, and how it fits into the Buf product.

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 4]
```diff
@@ -53,6 +54,19 @@ message OrganizationUser {
   OrganizationRoleSource organization_role_source = 4;
 }
 
+// UserPluginPreference is a nested structure that contains the
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/user.proto:57]: What is nested in this context?

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 5]
```diff
@@ -53,6 +54,19 @@ message OrganizationUser {
...
+// users preferred or selected language as well as a list of plugins
+// selected for that language
+message UserPluginPreference {
+  // The id of the user
+  string user_id = 1;
+  // The selected language that the user has indicidated
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/user.proto:63]: I'm not sure what a preferred language for plugins is. What is this used for? How does this fit into the Buf product?

Comment (gilwong00) [reply proto/buf/alpha/registry/v1alpha1/user.proto:63]: This is for the new Generated SDKs tab. We changed the flow so a user first selects their preferred language and then we display a list of plugins pertaining to that language. We want to persist the language and the plugins the user selected so the next time they visit this page we dont need to have them go through the selection process again

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 6]
```diff
@@ -105,6 +119,12 @@ service UserService {
   }
   // UpdateUserSettings update the user settings including description.
   rpc UpdateUserSettings(UpdateUserSettingsRequest) returns (UpdateUserSettingsResponse);
+  // GetUserPluginPreferences gets plugin preferences for the user.
```
[the remark this answers is not in the document]
Comment (gilwong00) [reply proto/buf/alpha/registry/v1alpha1/user.proto:122]: plugin preferences is the language and the list of plugins a user has selected in the new Generated SDKs tab

File: proto/buf/alpha/registry/v1alpha1/user.proto (hunk 5 above)
Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/user.proto:61]: The point here is that any context should be added to the docs - I should be able to read these docs and understand immediately what is going on here.

File: proto/buf/alpha/registry/v1alpha1/user.proto (hunk 6 above)
[the remark this answers is not in the document]
Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/user.proto]: Should be in docs

File: proto/buf/alpha/registry/v1alpha1/user.proto (hunk 5 above)
Comment (gilwong00) [reply proto/buf/alpha/registry/v1alpha1/user.proto:61]: Understood. Added a explanation that this type is meant for Generated SDKs.

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 7]
```diff
@@ -53,6 +54,17 @@ message OrganizationUser {
   OrganizationRoleSource organization_role_source = 4;
 }
 
+// UserPluginPreference contains the users selected language as well as a list of plugins selected for that language.
+// This is used for Generate SDKs where the user can choose a language and corresponding plugins to be persisted.
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/user.proto:58]: I still don't know what this means - what does it mean for a plugin to be persisted? What does it mean to choose a language?

I'd check out https://github.com/bazelbuild/remote-apis/blob/main/build/bazel/remote/asset/v1/remote_asset.proto as a good example of Protobuf API documentation.

**Review by bufdev — APPROVED** [review]

**Pull request merged**